js/widgets/forms/clearButton.js

```diff
@@ -24,55 +27,75 @@ define( [
 				this.options.clearBtn = true;
 			}
 
-			if ( !!this.options.clearBtn && this.inputNeedsWrap ) {
-				this._addClearBtn();
+			// We do nothing on startup if the options is off or if this is not a wrapped input
+			if ( !this.options.clearBtn || !this.inputNeedsWrap ) {
+				return;
+			}
+
+			if ( this.options.enhanced ) {
+				this._clearButton = this._outer.children( "a.ui-textinput-clear-button" );
+				this._clearButtonIcon = this._clearButton
+					.children( "span.ui-textinput-clear-button-icon" );
+				this._toggleClasses( "add" );
+				this._bindClearEvents();
+			} else {
+				this._addClearButton();
 			}
 		},
 
 		clearButton: function() {
-			return $( "<a href='#' tabindex='-1' aria-hidden='true' " +
-				"class='ui-input-clear ui-button ui-icon-delete ui-button-icon-only ui-corner-all'>" +
-				"</a>" )
+			var icon = $( "<span>" ),
+				button = $( "<a href='#' tabindex='-1' aria-hidden='true'></a>" )
 					.attr( "title", this.options.clearBtnText )
-					.text( this.options.clearBtnText );
+					.text( this.options.clearBtnText )
+					.append( icon );
+
+			return {
+				_clearButton: button,
+				_clearButtonIcon: icon
+			};
 		},
 
-		_clearBtnClick: function( event ) {
+		_clearButtonClick: function( event ) {
 			this.element.val( "" )
 					.focus()
 					.trigger( "change" );
-
-			this._clearBtn.addClass( "ui-input-clear-hidden" );
 			event.preventDefault();
 		},
 
-		_addClearBtn: function() {
-
-			if ( !this.options.enhanced ) {
-				this._enhanceClear();
-			}
-
-			$.extend( this, {
-				_clearBtn: this.widget().find("a.ui-input-clear")
-			});
+		_toggleClasses: function( operation ) {
+			this[ "_" + operation + "Class" ]( this._outer, "ui-textinput-has-clear-button" );
+			this[ "_" + operation + "Class" ]( this._clearButton, "ui-textinput-clear-button",
+				"ui-button ui-button-icon-only ui-button-right" );
+			this[ "_" + operation + "Class" ]( this._clearButtonIcon,
+				"ui-textinput-clear-button-icon",
+				"ui-icon-delete ui-icon" );
+		},
 
+		_addClearButton: function() {
+			this._addClass( this._outer, "ui-textinput-has-clear-button" );
+			$.extend( this, this.clearButton() );
+			this._toggleClasses( "add" );
+			this._clearButton.appendTo( this._outer );
 			this._bindClearEvents();
-
 			this._toggleClear();
-
 		},
 
-		_enhanceClear: function() {
-
-			this.clearButton().appendTo( this.widget() );
-			this.widget().addClass( "ui-input-has-clear" );
-
+		_removeClearButton: function( isDestroy ) {
+			if ( !isDestroy ) {
+				this._toggleClasses( "remove" );
+				this._unbindClearEvents();
+			}
+			this._clearButton.remove();
+			clearTimeout( this._toggleClearDelay );
+			delete this._clearButton;
```

Comment:
I don't think this is needed. If they are turning the option off they may turn it on and then all of these could be reused. if this is a true destroy the factory will take care of this. We don't delete widget properties in other places when toggling options same with the below.